optaplanner-core/src/main/java/org/optaplanner/core/config/util/ConfigUtils.java

```diff
@@ -390,7 +391,7 @@ public static int resolvePoolSize(String propertyName, String value, String... m
         }
     }
 
-    public static <C> MemberAccessor findPlanningIdMemberAccessor(Class<C> clazz) {
+    public static <C> MemberAccessor findPlanningIdMemberAccessor(DomainAccessType domainAccessType, Class<C> clazz) {
```

Comment:
Let's make domainAccessType the last parameter, because when reading code that calls this method, the clazz (= the what) is more important than the accesstype (the how)